release/ray_release/scripts/ray_bisect.py

```diff
@@ -19,32 +19,52 @@
 @click.argument("passing_commit", required=True, type=str)
 @click.argument("failing_commit", required=True, type=str)
 def main(test_name: str, passing_commit: str, failing_commit: str) -> None:
+    test = _get_test(test_name)
+    pre_sanity_check = _sanity_check(test, passing_commit, failing_commit)
+    if not pre_sanity_check:
+        logger.info(
+            "Failed pre-saniy check, the test might be flaky or fail due to"
+            " an external (not a code change) factors"
+        )
+        return
     commit_lists = _get_commit_lists(passing_commit, failing_commit)
-    blamed_commit = _bisect(test_name, commit_lists)
+    blamed_commit = _bisect(test, commit_lists)
     logger.info(f"Blamed commit found for test {test_name}: {blamed_commit}")
 
 
-def _bisect(test_name: str, commit_list: List[str]) -> str:
-    test = _get_test(test_name)
+def _bisect(test: Test, commit_list: List[str]) -> str:
     while len(commit_list) > 2:
         logger.info(
             f"Bisecting between {len(commit_list)} commits: "
             f"{commit_list[0]} to {commit_list[-1]}"
         )
         middle_commit_idx = len(commit_list) // 2
         middle_commit = commit_list[middle_commit_idx]
-        is_passing = _run_test(test, middle_commit)
+        is_passing = _run_test(test, [middle_commit])[middle_commit] == "passed"
         if is_passing:
             commit_list = commit_list[middle_commit_idx:]
         else:
             commit_list = commit_list[: middle_commit_idx + 1]
     return commit_list[-1]
 
 
-def _run_test(test: Test, commit: str) -> bool:
-    logger.info(f'Running test {test["name"]} on commit {commit}')
-    _trigger_test_run(test, commit)
-    return _obtain_test_result(commit)
+def _sanity_check(test: Test, passing_revision: str, failing_revision: str) -> bool:
```

Comment:
could you add a doc string on what the return value is?

I was a bit confused when that `pre_sanity_check = _sanity_check()`..